registry/src/model/mod.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 Helsing GmbH
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use atmosphere::prelude::*;
+use chrono::{DateTime, Utc};
+
+use serde::{Deserialize, Serialize};
+
+/// A user of the registry
+#[derive(Schema, Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[table(schema = "public", name = "users")]
+pub struct User {
+    #[sql(pk)]
+    pub id: i32,
+    #[sql(unique)]
+    pub handle: String,
+    #[sql(timestamp = created)]
+    //#[hook(insert = DateTime::now())]
+    pub created_at: DateTime<Utc>,
+    #[sql(timestamp = updated)]
+    //#[hook(update = DateTime::now())]
+    pub updated_at: DateTime<Utc>,
+    #[sql(timestamp = deleted)]
+    //#[hook(delete = DateTime::now())]
+    pub deleted_at: Option<DateTime<Utc>>,
+}
+
+/// A token that is identifying a user / used for authentication
+#[derive(Schema, Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[table(schema = "public", name = "user_tokens")]
+pub struct UserToken {
+    #[sql(pk)]
+    pub hash: String,
```

Comment:
What kind of hash is this/what is it used for? A JWT signature?

Reply:
cc @xfbs

Reply:
argon2 hash (params noted) of the randomly generated token

Reply:
PHC format?